pandas/core/arrays/base.py

```diff
@@ -666,6 +666,9 @@ def _values_for_argsort(self) -> np.ndarray:
         # Note: this is used in `ExtensionArray.argsort/argmin/argmax`.
         return np.array(self)
 
+    def _mask_for_argsort(self) -> np.ndarray:
+        return np.asarray(isna(self))
+
```

Comment:
Similar comment here, probably better to keep this logic in `SparseArray` instead of adding another method to the `ExtensionArray` interface